crates/indexer/src/indexer.rs

```diff
@@ -87,364 +104,415 @@ impl Web3Indexer {
         }
-        Ok(web3_tx_with_logs_vec)
+        Ok(None)
+    }
+
+    async fn batch_from_script(
+        &self,
+        txs: &[L2Transaction],
+    ) -> Result<std::collections::HashMap<u32, Option<Script>>> {
+        let from_ids = txs
+            .iter()
+            .map(|tx| {
+                let from_id: u32 = tx.raw().from_id().unpack();
+                from_id
+            })
+            .collect::<Vec<_>>();
+        let from_ids_len = from_ids.len();
+
+        let mut to_ids = txs
+            .iter()
+            .map(|tx| {
+                let to_id: u32 = tx.raw().to_id().unpack();
+                to_id
+            })
+            .collect::<Vec<_>>();
+        let to_ids_len = to_ids.len();
+
+        let mut ids = from_ids;
+        ids.append(&mut to_ids);
+
+        let id_sets: HashSet<u32> = std::collections::HashSet::from_iter(ids.into_iter());
+        let ids = Vec::from_iter(id_sets.into_iter());
+        let ids_len = ids.len();
+
+        log::debug!(
+            "batch_from_script request, from_id len: {}, to_id len: {}, id sets len: {}",
+            from_ids_len,
+            to_ids_len,
+            ids_len
+        );
+
+        let scripts = batch_account_id_to_script(&self.godwoken_async_client, ids.clone()).await?;
+
+        let mut hashmap = HashMap::<u32, Option<Script>>::new();
+        scripts.into_iter().enumerate().for_each(|(i, value)| {
+            let id = ids[i];
+            hashmap.insert(id, value);
+        });
+
+        Ok(hashmap)
+    }
+
+    async fn insert_l2block(&self, l2_block: L2Block) -> Result<(usize, usize)> {
+        let block_number = l2_block.raw().number().unpack();
+        let block_hash: gw_common::H256 = blake2b_256(l2_block.raw().as_slice()).into();
+        // let mut cumulative_gas_used: u128 = 0;
+        let l2_transactions = l2_block.transactions();
+        let l2_transactions_vec: Vec<L2Transaction> = l2_transactions.into_iter().collect();
+
+        let mut logs_len: usize = 0;
+        let mut web3_txs_len: usize = 0;
+
+        let id_script_hashmap = self.batch_from_script(&l2_transactions_vec).await?;
+
+        let txs_slice = l2_transactions_vec
+            .into_iter()
+            .chunks(TX_BATCH_SIZE)
+            .into_iter()
+            .map(|chunk| chunk.collect())
+            .collect::<Vec<Vec<_>>>();
+
+        // begin db transaction
+        let pool = &*POOL;
+        let mut pg_tx = pool.begin().await?;
+
+        let mut tx_index_cursor: u32 = 0;
+
+        let mut cumulative_gas_used: u128 = 0;
+        let mut total_gas_limit: u128 = 0;
+        let mut total_gas_used: u128 = 0;
+        for txs in txs_slice {
+            let l2_transaction_with_logs_vec = txs
+                .into_par_iter()
+                .map(|tx| {
+                    self.filter_single_transaction(
+                        tx,
+                        block_number,
+                        block_hash,
+                        0,
```

Comment:
`tx_index` always be `0`?

Reply:
Tx index unknown here, should update later: https://github.com/nervosnetwork/godwoken-web3/blob/refactor-sync/crates/indexer/src/indexer.rs#L485

Reply:
So this parameter 'tx_index' seems useless, is it? Let's remove this parameter from the function declaration?

Reply:
Updated